test/integration/Microservice.test.js

```diff
@@ -43,18 +42,36 @@ describe('The Microservice', function() {
         expect(service)
             .to.have.property('api')
             .that.has.property('base')
-            .that.equals('http://0.0.0.0:3333/api');
+            .that.equals('http://0.0.0.0:3333/api/');
     });
 
     it('the api client allows querying the api', async function(){
+
         const service = await this.ms.start();
-        const response = await service.api
+        const response = await service
+            .api
             .get('/tests')
-            .set('accept', 'application/json')
-            .then((response) => response);
+            .set('accept', 'application/json');
+
         expect(response).to.have.property('status', 200);
     });
 
+    it('exposes an error handler which enriches the error with the serviceName ' +
+        'if configured accordingly', async function(){
+
+        const service = await this.ms.start();
+        try {
+            const response = await service
+                .api
+                .get('/nonExisting')
+                .set('accept', 'application/json');
+            return Promise.reject(new Error('Request to non existing route should fail.'))
```

Comment:
Why do you need to reject the promise manually here? Shouldn't the request reject it?

Reply:
Yeah, it should, but if it does not, the test will not fail (and if i throw it, it would be caught by the catch clause and give me an error message which does not tell me why the test failed).